### ISSUE TITLE
Are paths longer than 255 characters supported?

### ISSUE THREAD

**Issue by daarsan** [jborean93/smbprotocol#303]
Hi all,
I'm afraid this is not an issue but a question to know if this is supported or if there is a workaround to access (extremely) long paths.
Let's say I have to copy files to a network folder and the folder where the file is supposed to be is: 
`\\myserver.domain.org\D$\MigrationFiles\Test\Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aenean eget felis urna. Aliquam tellus diam, rhoncus non faucibus vitae, lacinia dapibus nulla. Cras egestas, arcu eu eleifend maximus, felis enim aliquet erat, id iaculis metus nulla sit amet nunc. In quis tincidunt nisi. Phasellus nibh elit, molestie eu vehicula eget, mattis a sem. Quisque varius iaculis justo, non cursus mi tristique ac. Phasellus hendrerit`

In principle I could do: 
```
 smbclient.mkdir(r"\\myserver.domain.org\D$\MigrationFiles\Test\Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aenean eget felis urna. Aliquam tellus diam, rhoncus non faucibus vitae, lacinia dapibus nulla. Cras egestas, arcu eu eleifend maximus, felis enim aliquet erat, id iaculis metus nulla sit amet nunc. In quis tincidunt nisi. Phasellus nibh elit, molestie eu vehicula eget, mattis a sem. Quisque varius iaculis justo, non cursus mi tristique ac. Phasellus hendrerit")
```
But I get following exception:
`smbprotocol.exceptions.SMBOSError: [Error 0] [NtStatus 0xc0000033] Unknown NtStatus error returned 'STATUS_OBJECT_NAME_INVALID': `

I read that Windows can support long paths using the prefix "\\?\UNC\" but in this case it does not help.
Any idea or suggestion about how can I create and list directories like this one?

Best regards.

**Comment by jborean93** [comment]
There is nothing in the MS-SMB2 protocol docs that would indicate there is any special handling for long paths. When testing it out locally I can create paths that in when put together exceed the `MAX_PATH` value just fine.

Keep in mind the maximum length of an individual path component (filenames/dirs in the path) is 255 characters for NTFS drives so while you can have the full path exceed 255 characters, an individual file or directory cannot. This is a hard limitation of NTFS and Windows itself cannot create such a path even with the `\\?\` prefix.

**Comment by daarsan** [comment]
In that case, I will need to preprocess paths before I try to create new files and folders. I was surprised when I saw that users created such long folders in the source system and I was wondering if I was able to have the same somewhere else.

Just for the record, I was able to create the folder using Windows Explorer but not using Powershell or cmd (even enabling long path support: https://learn.microsoft.com/en-us/windows/win32/fileio/maximum-file-path-limitation?tabs=registry).

In any case, thank you very for your help!
